src/host/main.cpp

```diff
@@ -634,6 +639,20 @@ int main(int argc, char** argv)
     ccf_config.joining.target_port = target_rpc_address.port;
     ccf_config.joining.network_cert = files::slurp(network_cert_file);
     ccf_config.joining.join_timer = join_timer;
+
+    auto snapshot_file = snapshots.find_latest_snapshot();
+    if (snapshot_file.has_value())
+    {
+      ccf_config.joining.snapshot = files::slurp(snapshot_file.value());
+      LOG_INFO_FMT(
+        "Found latest snapshot file: {} (size: {})",
+        snapshot_file.value(),
+        ccf_config.joining.snapshot.size());
+    }
+    else
+    {
+      LOG_INFO_FMT("No snapshot found, node will join from genesis");
```

Comment:
```suggestion
      LOG_INFO_FMT("No snapshot found, node will request transactions from the beginning");
```

I suggest we generally drop the term `genesis` everywhere we can. We've made a number of changes to have the first transaction be less special, and that's our direction of travel.